gdpr_assist/management/commands/anonymise_db.py

```diff
@@ -39,8 +137,30 @@ def handle(self, *args, **options):
             confirm = "yes"
 
         if confirm == "yes":
-            for model in registry.models_allowed_to_anonymise():
-                model.objects.all().anonymise()
+
+            # Begin applying the requested anonymisation strategy to each model
+            for model in self._models_for_anonymisation():
+                category = StrategyHelper.category_for_model(model)
+                strategy = strategies.get(category)
+
+                if strategy == StrategyHelper.STRATEGY_ANONYMISE:
+                    if issubclass(model, PrivacyModel) and model.check_can_anonymise():
+                        model.objects.all().anonymise()
+                    else:
+                        raise CommandError(
+                            """Cannot anonymise {} model {}!""".format(category, model),
+                            """Please check your anonymisation settings."""
+                        )
+
+                elif strategy == StrategyHelper.STRATEGY_DELETE:
+                    model.objects.all().delete()
```

Comment:
This line seems to be a challenge for deletion of large database tables.  Django loads each of the models into memory during deletion (to inspect for signals and cascading relationships), and that can lead to memory exhaustion.

Reply:
...and there seem to be at least two approaches to help resolve this:

- Iterating over the results of a queryset in a loop, and performing regular garbage collection at some interval
- Using Django's built-in (albeit private-by-convention) `_raw_delete` method

There's some relevant context and history about bulk deletion support in [this Django ticket](https://code.djangoproject.com/ticket/9519).

In simplified terms, the former approach is slow-but-correct, and the latter is fast but risks behavioural inconsistencies (Django signals and deletion cascades are dropped).